flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-elasticsearch/src/test/java/org/apache/flink/cdc/connectors/elasticsearch/sink/ElasticsearchDataSinkITCaseTest.java

```diff
@@ -0,0 +1,313 @@
+        return new ElasticsearchContainer(ELASTICSEARCH_IMAGE)
+                .withEnv("discovery.type", "single-node")
+                .withEnv("xpack.security.enabled", "false")
+                .withEnv("ES_JAVA_OPTS", "-Xms2g -Xmx2g")
+                .withEnv("logger.org.elasticsearch", "ERROR")
+                .withLogConsumer(new Slf4jLogConsumer(LOG))
+                .waitingFor(Wait.forListeningPort().withStartupTimeout(Duration.ofMinutes(5)));
+    }
+
+    private ElasticsearchClient createElasticsearchClient() {
+        RestClientTransport transport =
+                new RestClientTransport(
+                        RestClient.builder(
+                                        new HttpHost(
+                                                ELASTICSEARCH_CONTAINER.getHost(),
+                                                ELASTICSEARCH_CONTAINER.getFirstMappedPort(),
+                                                "http"))
+                                .build(),
+                        new JacksonJsonpMapper());
+        return new ElasticsearchClient(transport);
+    }
+
+    private void runJobWithEvents(List<Event> events) throws Exception {
+        ElasticsearchSinkOptions options = createSinkOptions();
+        StreamExecutionEnvironment env = createStreamExecutionEnvironment();
+        ElasticsearchDataSink<Event> sink =
+                new ElasticsearchDataSink<>(options, ZoneId.systemDefault());
+
+        DataStream<Event> stream = env.fromCollection(events, TypeInformation.of(Event.class));
+        Sink<Event> elasticsearchSink = ((FlinkSinkProvider) sink.getEventSinkProvider()).getSink();
+        stream.sinkTo(elasticsearchSink);
+
+        env.execute("Elasticsearch Sink Test");
+    }
+
+    private ElasticsearchSinkOptions createSinkOptions() {
+        NetworkConfig networkConfig =
+                new NetworkConfig(
+                        Collections.singletonList(
+                                new HttpHost(
+                                        ELASTICSEARCH_CONTAINER.getHost(),
+                                        ELASTICSEARCH_CONTAINER.getFirstMappedPort())),
+                        null,
+                        null,
+                        null,
+                        null,
+                        null);
+
+        return new ElasticsearchSinkOptions(
+                5, 1, 10, 50 * 1024 * 1024, 1000, 10 * 1024 * 1024, networkConfig);
+    }
+
+    private StreamExecutionEnvironment createStreamExecutionEnvironment() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        env.enableCheckpointing(3000);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        return env;
+    }
+
+    private void verifyInsertedData(
+            TableId tableId, String id, int expectedId, double expectedNumber, String expectedName)
```

Comment:
We can pass `List<String>` for column name and `List<Object>` for expected object to simplify verification.